vispy/visuals/gridlines.py

```diff
@@ -86,6 +109,30 @@ def __init__(self, scale=(1, 1), color='w'):
         self.shared_program.frag['get_data'] = self._grid_color_fn
         cfun = Function('vec4 null(vec4 x) { return x; }')
         self.shared_program.frag['color_transform'] = cfun
+        self.unfreeze()
+        self.bounds = bounds
```

Comment:
It appears the website building is failing on running an exampel because `bounds` is already used at the higher levels of vispy. We may need to come up with a better name for this keyword and property.

Reply:
huh I see, that's strange, does that mans `GridLines` is used as a nested visual, and the call to `x.bounds(...)` somehow get passed to `GridLines`?

Reply:
Visual classes are wrapped/subclassed with other classes to work with the `SceneCanvas`, but it looks like I was wrong about that being the issue. Turns out the `BaseVisual` class has a `bounds` method:

https://github.com/vispy/vispy/blob/664460aa6c8479b71eee9cdfc4edc048853d0764/vispy/visuals/visual.py#L241-L255

Yeah, I don't like `gridline_bounds`. `grid_bounds` maybe, but that's a last resort. Can we come up with anything better? `extents`? Eh that's kind of weird.

Reply:
I feel like `axis_limits` is used somewhere else too, but even if it isn't `axis` is a separate widget/visual and since it is not involved here it could be confusing if/when this visual is used in a plot alongside the AxisVisuals. I'd be open to the two separate properties if you like that. Matching matplotlib wouldn't be the worst and the two properties make it more obvious what the order of the elements are.

Bounds...extents...limits...something else? Not that my word should be final but now I'm starting to lean towards `grid_bounds` if we're going with a single property.

Reply:
I kinda feel like the most ergonomic would be to have independent kwargs and properties (`xmin`, `xmax`, `ymin`, `ymax`, or something like that), but also `xlimits` and `ylimits` makes sense. We can always add a convenience property for the whole `grid_bounds`.